products/bridge/test/test.ts

```diff
@@ -299,43 +317,46 @@ describe("Bridge", function () {
       success,
       response,
       nonce,
-      signerIndices,
       signatures
     );
 
-    const blockNum = await ethers.provider.getBlockNumber();
-    const logs = await validators1[signerIndices[0]].provider.getLogs({
-      fromBlock: blockNum - 100,
-      toBlock: blockNum,
-      address: caller,
-      topics: [ethers.id("Succeeded(uint256)")],
-    });
-    var res = ethers.AbiCoder.defaultAbiCoder().decode(
-      ["uint256"],
-      logs[0].data
-    );
-    console.log("Incremented", num, "to", ethers.toNumber(res[0]));
+    const filter = twin1.filters.Succeeded();
+    const logs = await twin1.queryFilter(filter);
```

Comment:
As discussed, it's essential to only consider events in finalized blocks, therefore the code before refactoring fetched `ethers.provider.getBlockNumber('finalized')` and queried only until that block number.